lib/oplog/tail.go

```diff
@@ -514,7 +508,19 @@ func (tailer *Tailer) parseRawOplogEntry(entry rawOplogEntry, txIdx *uint) []opl
 		if out.Operation == operationUpdate {
 			out.DocID = entry.Update.ID
 		} else {
-			out.DocID = data["_id"]
+			idLookup := entry.Doc.Lookup("_id")
+			oid, ok := idLookup.ObjectIDOK()
+			if ok {
+				out.DocID = oid.String()
+			} else {
+				oidString, ok := idLookup.StringValueOK()
+				if ok {
+					out.DocID = oidString
+				} else {
+					log.Log.Errorf("failed to unmarshal objectId")
+					return nil
```

Comment:
Should this raise an error instead of returning nil? I realize this gets papered over in processOplogEntry, but this seems like a more natural place to raise the error then let it bubble up. Otherwise you can't distinguish the case of `_id was set to nil` from `_id was set to something other than ObjectID or String`

Reply:
Distinguishing the error cause makes sense, I split the error into two now: empty and wrong type.  But returning nil is to follow the existing pattern and skip this entry as unprocessable, logging the error.  The function doesn't currently return an error, so adding that would be a refactor which seems out of scope for this PR in my opinion.

Reply:
The reason that I'm nitpicking this is that this is a sharp edge for a future dev. i.e., they need to "just know" that _id = null means something special, not that _id is literally null. Maybe leave a comment at least?

Reply:
It won't return the object with _id = null if there's an error, rather it returns `nil`.  If there is no error, it returns `[]oplogEntry{out}`.  So it won't just continue with a null _id, which is why this seems safe to me.

Reply:
Oh I see, sorry, got it